numba/targets/arraymath.py

```diff
@@ -999,3 +1019,192 @@ def dot_3(context, builder, sig, args):
+        with builder.if_else(invalid_arg) as (then, otherwise):
+            raise_err = context.call_conv.return_user_exc
+            with then:
+                raise_err(builder, ValueError,
+                          ('One argument passed to getrf is invalid',)
+                          )
+            with otherwise:
+                raise_err(builder, ValueError,
+                          ('Matrix is singular and cannot be inverted',)
+                          )
+
+    # Compute the optimal lwork.
+    lwork = make_constant_slot(context, builder, types.intc, -1)
+    work = cgutils.alloca_once(builder, context.get_value_type(xty.dtype))
+    call_xxgetri(context, builder, xty, x_shapes, o.data, i.data, work,
+                 lwork, info)
+
+    info_val = builder.load(info)
+    lapack_error = builder.icmp_signed('!=', info_val, zero)
+
+    with builder.if_then(lapack_error, False):
+        raise_err = context.call_conv.return_user_exc
+        raise_err(builder, ValueError,
+                  ('One argument passed to getri is invalid',)
+                  )
+
+    # Allocate a work array of the optimal size as computed by getri.
+    def allocate_work(x, size):
+        """Allocate the work array.
+
+        """
+        size = int(1.01 * size.real)
+        return numpy.empty((size,), dtype=x.dtype)
+
+
+    wty = types.Array(dtype, 1, 'C')
+    work = context.compile_internal(builder, allocate_work,
+                                    signature(wty, xty, dtype),
+                                    (args[0], builder.load(work)))
+
+    w = make_array(wty)(context, builder, work)
+    w_shapes = cgutils.unpack_tuple(builder, w.shape)
+    lw, = w_shapes
+
+    builder.store(context.cast(builder, lw, types.intp, types.intc),
+                  lwork)
+
+    # Compute the matrix inverse.
+    call_xxgetri(context, builder, xty, x_shapes, o.data, i.data, w.data,
+                 lwork, info)
+
+    info_val = builder.load(info)
+    lapack_error = builder.icmp_signed('!=', info_val, zero)
+    invalid_arg = builder.icmp_signed('<', info_val, zero)
+
+    with builder.if_then(lapack_error, False):
+        with builder.if_else(invalid_arg) as (then, otherwise):
+            raise_err = context.call_conv.return_user_exc
+            with then:
+                raise_err(builder, ValueError,
+                          ('One argument passed to getri is invalid',)
+                          )
+            with otherwise:
+                raise_err(builder, ValueError,
+                          ('Matrix is singular and cannot be inverted',)
+                          )
+
+    return impl_ret_borrowed(context, builder, sig.return_type, out)
+
+
+@lower_builtin(numpy.linalg.inv, types.Array)
+def inv(context, builder, sig, args):
+    """
+    np.linalg.inv(a)
+    """
+    ensure_lapack()
+
+    ndims = sig.args[0].ndim
+    if ndims == 2:
+        return mat_inv(context, builder, sig, args)
+    else:
+        assert 0
```

Comment:
Probably want to raise a `ValueError` with a message about supported dimensions?